designer/server/src/plugins/routes/api.js

```diff
@@ -0,0 +1,92 @@
+import { Schema } from '@defra/forms-model'
+
+import * as forms from '~/src/lib/forms.js'
+
+/**
+ * @type {ServerRoute}
+ */
+export const getFormWithId = {
+  method: 'GET',
+  path: '/api/{id}/data',
+  options: {
+    async handler(request, h) {
+      const { id } = request.params
+      try {
+        const formJson = await forms.getDraftFormDefinition(id)
+
+        return h.response(/** @type {object} */ (formJson))
+      } catch (error) {
+        request.logger.error(['GET /api/{id}/data', 'getFormWithId'], error)
+        throw error
+      }
+    }
+  }
+}
+
+/**
+ * @type {ServerRoute}
+ */
+export const putFormWithId = {
+  method: 'PUT',
+  path: '/api/{id}/data',
+  options: {
+    payload: {
+      parse: true
+    },
+    async handler(request, h) {
+      const { id } = request.params
+      try {
+        const { value, error } = Schema.validate(request.payload, {
+          abortEarly: false
+        })
+
+        if (error) {
+          request.logger.error(
+            ['error', `/api/${id}/data`],
+            [error, request.payload]
+          )
+
+          throw new Error(`Schema validation failed, reason: ${error.message}`)
+        }
+
+        await forms.updateDraftFormDefinition(id, JSON.stringify(value))
+
+        return h.response({ ok: true }).code(204)
+      } catch (err) {
+        request.logger.error('Designer Server PUT /api/{id}/data error:', err)
+        const errorSummary = {
+          id,
+          payload: request.payload,
+          errorMessage: err.message,
+          error: err.stack
+        }
+        request.yar.set(`error-summary-${id}`, errorSummary)
+        return h.response({ ok: false, err }).code(500)
+      }
+    }
+  }
+}
+
+/**
+ * @type {ServerRoute}
```

Comment:
```suggestion
 * @type {ServerRoute<{ Payload: FormDefinition }>}
```

Reply:
Did you miss this one?

Payload was just a suggestion, but without it you'll get ESLint errors on lines 80 + 83

Leave it for now if you like, one for another PR

Reply:
Ah, no I didn't miss it, I hadn't created this endpoint so I wasn't planning to make changes to it. I'm not sure the payload is actually a FormDefinition, it's a logging endpoint to errors. I plan on getting some time to stripping out any dead code to remove our development burden anyway, so it can be looked at then.